src/plugins/discover/public/application/context/context_app_content.tsx

```diff
@@ -162,6 +169,13 @@ export function ContextAppContent({
             onRemoveColumn={onRemoveColumn}
             onSetColumns={onSetColumns}
             DocumentView={DiscoverGridFlyout}
+            services={{
+              fieldFormats,
+              addBasePath,
+              uiSettings,
+              dataViewFieldEditor,
+              toastNotifications,
+            }}
```

Comment:
To avoid unnecessary rerenderings (due to new objects), we could pass `services={services}` and typescript would check if all required services were provided. And at the top of the file we would have:

```
const services = useDiscoverServices();
const { uiSettings: config } = services;
```

```suggestion
            services={services}
```